README.md

```diff
@@ -71,50 +76,17 @@ Finalizers are tricky business and it is best to avoid them.  They can be invoke
-  return { memory, size, freeList, finalizationGroup };
-}
-
-function allocate(allocator) {
-  const { memory, size, freeList, finalizationGroup } = allocator;
-  if (freeList.length === 0) throw new RangeError('out of memory');
-  const index = freeList.shift();
-  const buffer = new Uint8Array(memory, index * size, size);
-  finalizationGroup.register(buffer, index);
-  return buffer;
-}
-```
-
-This code uses a few features of the `FinalizationGroup` API:
-- An object can have a finalizer referenced by calling the `register` method of `FinalizationGroup`. In this case, two arguments are passed to the `register` method:
-  - The object whose lifetime we're concerned with. Here, that's the `Uint8Array`
-  - A “holdings” value, which is used to represent that object when cleaning it up in the finalizer. In this case, the holdings are an integer corresponding to the offset within the `WebAssembly.Memory` object.
-- The `FinalizationGroup` constructor is called with a callback as an argument. This callback is called with an iterator of the holdings values.
+### Locating and responding to external resource leaks
 
-The finalizer callback is called *after* the object is garbage collected, a pattern which is sometimes called "post-mortem". For this reason, a separate "holdings" value is put in the iterator, rather than the original object--the object's already gone, so it can't be used.
+Finalizers can locate external resource leaks. For example, if an open file is garbage collected, the underlying operating system resource could be leaked. Although the OS will likely free the resources when the process exits, this sort of leak could make long-running processes eventually exhaust the number of file handles available. To catch these bugs, a `FinalizationGroup` can be used to log the existence of file objects which are garbage collected before being closed.
 
-The `FinalizationGroup` callback is passed an iterator of holdings to give that callback control over how much work it wants to process. The callback may pull in only part of the iterator, and in this case, the rest of the work would be "saved for later". The callback is not called during execution of other JavaScript code, but rather "in between turns"; it is currently proposed to be restricted to run after all of of the `Promise`-related work is done, right before turning control over to the event loop.
-
-### Releasing external resources as a backstop
-
-Another use case is running custom clean-up code after Finalization use case: e.g. file handle: run some code once a file is no longer in use, in case nothing points to it anymore. Note, it’s still encouraged to use an explicit method call to clean up the resource; this is just a backup.
+The `FinalizationGroup` class represents a group of objects registered with a common finalizer callback. This construct can be used to return the linear memory allocation to the free list when the related JavaScript object is garbage-collected.
```

Comment:
I'm not sure the free list example mention is the best here since right after is the File cleanup example